ChangeTests/ChangeUnitTests.cs

```diff
@@ -0,0 +1,96 @@
+namespace ChangeTests
+{
+    [TestClass]
+    public class ChangeUnitTests
+    {
+        private Dictionary<int, int> availableCoins;
+        MyCash myCash;
+
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            availableCoins = new Dictionary<int, int> {
+            { 100, 10 },
+            { 50, 20 },
+            { 10, 50 },
+            { 5, 100 },
+            { 2, 100 },
+            { 1, 100 }
+            };
+
+            myCash = new MyCash(availableCoins);
+        }
+
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void Calculate_Negative_value()
+        {
+
+            // Act
+            (bool isAvailable, Dictionary<int, int> coins) = myCash.CalculateChange(-10);
+
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentException))]
+        public void Calculate_zero_value() {
+            (bool isAvailable, Dictionary<int, int> coins) = myCash.CalculateChange(0);
+        }
+
+        [TestMethod]
+        public void Calculate_Positive_Value()
+        {
+
+            (bool isAvailable, Dictionary<int, int> coins) = myCash.CalculateChange(123);
+
+            // Assert
+            Assert.IsTrue(isAvailable);
+            
+        }
+
+        [TestMethod]
+        public void Calculate_From_Possitive_Value_Query() {
+            Dictionary<int, int> expected_change = new Dictionary<int, int> {
+                {100, 1},
+                {10, 2},
+                {2, 2}
+            };
+
+            (bool isAvailable, Dictionary<int, int> coins) = myCash.CalculateChange(124);
+            Assert.AreEqual(expected_change[100], coins[100]);
+            Assert.AreEqual(expected_change[10], coins[10]);
+            Assert.AreEqual(expected_change[2], coins[2]);
+        }
+
+
+        [TestMethod]
+        [ExpectedException (typeof(ArgumentException))]
+        public void Calculate_OverFLow_Value()
+        {
+            (bool isAvailable, Dictionary<int, int> coins) = myCash.CalculateChange(3301);
```

Comment:
This puzzled me at first. Our salesman asks our system if $3,301 can be returned having current coins in the checkout point. If it is not possible it is a normal (not exceptional) case and we should return false. And that is the case in the contract of your method with isAvailable flag. However, you never return false in your code. You substituted it with exceptions which is considered as an anti-pattern. The first thing I googled about that: https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why. You can check out more and try to address this issue.

Reply:
Done. Now func returns false and empty dictionary.